Source/WebCore/dom/SlotAssignment.h

```diff
@@ -135,6 +137,35 @@ class NamedSlotAssignment : public SlotAssignment {
 #endif
 };
 
+class ManualSlotAssignment : public SlotAssignment {
+public:
+    ManualSlotAssignment() = default;
+
+    HTMLSlotElement* findAssignedSlot(const Node&) final;
+
+    const Vector<WeakPtr<Node>>* assignedNodesForSlot(const HTMLSlotElement&, ShadowRoot&) final;
+    void renameSlotElement(HTMLSlotElement&, const AtomString&, const AtomString&, ShadowRoot&) final;
+    void addSlotElementByName(const AtomString&, HTMLSlotElement&, ShadowRoot&) final;
+    void removeSlotElementByName(const AtomString&, HTMLSlotElement&, ContainerNode*, ShadowRoot&) final;
+    void slotManualAssignmentDidChange(HTMLSlotElement&, Vector<WeakPtr<Node>>& previous, Vector<WeakPtr<Node>>& current, ShadowRoot&) final;
+    void slotFallbackDidChange(HTMLSlotElement&, ShadowRoot&) final;
+
+    void hostChildElementDidChange(const Element&, ShadowRoot&) final;
+    void hostChildElementDidChangeSlotAttribute(Element&, const AtomString&, const AtomString&, ShadowRoot&) final;
+
+    void willRemoveAssignedNode(const Node&) final;
+    void didRemoveAllChildrenOfShadowHost(ShadowRoot&) final;
+    void didMutateTextNodesOfShadowHost(ShadowRoot&) final;
+
+private:
+    struct Slot {
+        Vector<WeakPtr<Node>> cachedAssignment;
+        uint32_t cachedVersion { 0 };
+    };
+    WeakHashMap<HTMLSlotElement, Slot> m_slots;
+    uint32_t m_slottableVersion { 0 };
```

Comment:
These are probably not going to overrun but there is probably no reason not to just use uin64_t either.

Reply:
Ok, I'd change it to uint64_t.